crates/api/src/builder/api.rs

```diff
@@ -378,6 +370,225 @@ where
         Ok(StatusCode::OK)
     }
 
+    /// Handles the submission of a new block with inclusion proofs.
+    ///
+    /// This function extends the `submit_block` functionality to also handle inclusion proofs:
+    /// 1. Receives the request and decodes the payload into a `SignedBidSubmission` object.
+    /// 2. Validates the builder and checks against the next proposer duty.
+    /// 3. Verifies the signature of the payload.
+    /// 4. Fetches and handles inclusion proofs.
+    /// 5. Runs further validations against the auctioneer.
+    /// 6. Simulates the block to validate the payment.
+    /// 7. Saves the bid and proofs to the auctioneer and db.
+    ///
+    /// Implements this API: <https://flashbots.github.io/relay-specs/#/Builder/submitBlockWithProofs>
+    pub async fn submit_block_with_proofs(
+        Extension(api): Extension<Arc<BuilderApi<A, DB, S, G>>>,
+        req: Request<Body>,
+    ) -> Result<StatusCode, BuilderApiError> {
+        let request_id = Uuid::new_v4();
+        let mut trace = SubmissionTrace { receive: get_nanos_timestamp()?, ..Default::default() };
+        let (head_slot, next_duty) = api.curr_slot_info.read().await.clone();
+
+        info!(
+            request_id = %request_id,
+            event = "submit_block_with_proofs",
+            head_slot = head_slot,
+            timestamp_request_start = trace.receive,
+        );
+
+        // Decode the incoming request body into a payload with proofs
+        let (payload, is_cancellations_enabled) = 
+            decode_payload(req, &mut trace, &request_id).await?;
+        let block_hash = payload.message().block_hash.clone();
+
+        // Verify that we have a validator connected for this slot
+        if next_duty.is_none() {
+            warn!(request_id = %request_id, "could not find slot duty");
+            return Err(BuilderApiError::ProposerDutyNotFound);
+        }
+        let next_duty = next_duty.unwrap();
```

Comment:
This is more idiomatic:
```suggestion
        let Some(next_duty) = next_duty else {
            warn!(request_id = %request_id, "could not find slot duty");
            return Err(BuilderApiError::ProposerDutyNotFound);
        }
```

Reply:
Thanks for the suggestion I agree, I copied the already written code of `submit_block` where they did it this way. I am trying to not make extra changes and keep the coding style same.